ARCHITECTURE.md

```diff
@@ -0,0 +1,15 @@
+# DocumentViewer Architecture
+
+DocumentViewer catalogs documents in S3, and allows users to query and view them.
+
+## Catalog
+
+The S3 bucket storing the documents includes metadata files which list every file present, along with information about each file, such as the name and date of birth of the rider the document relates to. When the DocumentViewer application starts up, the `catalog` gets all metadata files in the S3 bucket and reads them to build an internal catalog of all files in the bucket. This data is held in-memory using an ETS table.
```

Comment:
question: Not sure this is the right place, but: how are these metadata files created? When they're updated, how does the Catalog update the in-memory DB?